### PR TITLE
[react-relay] Type-check fragment references.

### PR DESCRIPTION
**Pull request by alloy**
Related to https://github.com/relay-tools/relay-compiler-language-typescript/issues/7

### CODE REVIEW

**Comment by alloy** [comment]
@kastermester I now see you’re not in the list of maintainers, you should add yourself.

**Comment by alloy** [comment]
@kastermester I didn’t do the `InstanceType` part from your suggestion yet, because it was a lot of changes and I rather would do this iteratively. As far as I could tell that was all related to typing the `componentRef` prop anyways and stuff around that is changing in Relay to use `React.forwardRef`, so I figured we might as well revisit that.

File: types/react-relay/index.d.ts [hunk 1]
```diff
@@ -25,15 +26,22 @@ import * as RelayRuntimeTypes from "relay-runtime";
 
 // Taken from https://github.com/pelotom/type-zoo
 // tslint:disable-next-line:strict-export-declare-modifiers
-type Omit<T, K extends keyof T> = Pick<T, ({ [P in keyof T]: P } & { [P in K]: never } & { [x: string]: never, [x: number]: never })[keyof T]>;
+type Omit<T, K extends keyof T> = Pick<
+    T,
+    ({ [P in keyof T]: P } & { [P in K]: never } & { [x: string]: never; [x: number]: never })[keyof T]
+>;
```
Comment (ckknight) [inline types/react-relay/index.d.ts]: Why not use `Pick<T, Exclude<keyof T, K>>`?

File: types/react-relay/index.d.ts [hunk 2]
```diff
@@ -96,10 +104,16 @@ export class QueryRenderer extends ReactRelayQueryRenderer {}
 // ~~~~~~~~~~~~~~~~~~~~~
 // createFragmentContainer
 // ~~~~~~~~~~~~~~~~~~~~~
-export function createFragmentContainer<T>(
-    Component: React.ComponentType<T>,
+
+// export function createFragmentContainer<TContainer extends React.ComponentType<any>>(
+//     Component: TContainer,
+//     fragmentSpec: RelayRuntimeTypes.GraphQLTaggedNode | GeneratedNodeMap,
+// ): React.ComponentClass<ContainerProps<ConstructorProps<TContainer>, TContainer>>;
```
Comment (ckknight) [inline types/react-relay/index.d.ts]: Do you want to leave this comment in?

File: types/react-relay/index.d.ts (hunk 1 above)
Comment (alloy) [reply types/react-relay/index.d.ts]: I was actually doing that right now indeed 👍

File: types/react-relay/index.d.ts (hunk 2 above)
Comment (alloy) [reply types/react-relay/index.d.ts]: Nope, good catch!

**Comment by alloy** [comment]
@ckknight Done, thanks!

**Review by ckknight — APPROVED** [review]
Looks great! 👍🏻

**Pull request merged**